### ISSUE TITLE
Code completion error

### ISSUE THREAD

**Issue by alexreg** [carols10cents/rust.tmbundle#16]
I press the shortcut for code completion (by the way, why isn't it something simpler like Tab?) and TextMate freezes up for about 60 seconds, then gives me the following error message:

```
/usr/local/bin/multirust: fork: Resource temporarily unavailable
````

What's going on here?

NB: my environment consists of a simple script `/usr/local/bin/racer` which contains:
```bash
#!/bin/bash

multirust run override racer "$@"
```

**Comment by carols10cents** [comment]
> I press the shortcut for code completion (by the way, why isn't it something simpler like Tab?)

I'm not sure, you'll have to ask the Textmate developers that -- [all Textmate bundles use esc for completion](http://manual.macromates.com/en/working_with_text#completion).

Have you set the RUST_SRC_PATH variable in your textmate preferences?

I've been meaning to set up multirust anyway, I'll see if I can reproduce.

**Comment by alexreg** [comment]
Fair enough. It seems to be Alt+Esc on my system.

Yeah, do please let me know how it works for you, with multirust.

**Comment by infininight** [comment]
@alexreg The tab key can't be used for completion because it triggers snippets and of course inserting a tab character. Escape was chosen as the next best for completion, and option-escape is the secondary if two completion commands need to exist simultaneously.

**Comment by alexreg** [comment]
That’s fair enough, I suppose. Other editors (e.g. Visual Studio) manage to differentiation indentation tabs from tab-completion tabs, but the snippets feature does complicate things.

**Comment by carols10cents** [comment]
Hi @alexreg, I'm sorry I've taken so long to dig into this. I'm able to get the hanging behavior you describe using only the command line, without textmate... could you take a look at what I'm doing and let me know if you can reproduce what I'm seeing, and if not, what is different between your environment and mine?

I have this multirust installed:

```
$ multirust --version
multirust 0.7.0 (f3974f2b9 2015-12-14 14:49:29 +0000)
```

I have the rust source in /Users/carolnichols/rust/rust/src and RUST_SRC_PATH set in my terminal env:

```
$ echo $RUST_SRC_PATH
/Users/carolnichols/rust/rust/src
```

I used `cargo install` in multirust beta to install racer:

```
$ multirust run beta cargo install racer
...
$ which racer
/Users/carolnichols/.multirust/toolchains/beta/cargo/bin/racer
```

I can get some completions from racer on the command line directly using `multirust run override` at this point:

```
$ multirust run override racer complete std::io::B
MATCH BufReader,47,11,/Users/carolnichols/rust/rust/src/libstd/io/buffered.rs,Struct,pub struct BufReader<R> {
MATCH BufWriter,299,11,/Users/carolnichols/rust/rust/src/libstd/io/buffered.rs,Struct,pub struct BufWriter<W: Write> {
MATCH BufRead,1252,10,/Users/carolnichols/rust/rust/src/libstd/io/mod.rs,Trait,pub trait BufRead: Read {
MATCH Broadcast,1479,11,/Users/carolnichols/rust/rust/src/libstd/io/mod.rs,Struct,pub struct Broadcast<T, U> {
MATCH Bytes,1612,11,/Users/carolnichols/rust/rust/src/libstd/io/mod.rs,Struct,pub struct Bytes<R> {
```

Then I put your script on my path, with the name `script-racer` for the moment:

```
$ which script-racer
/Users/carolnichols/bin/script-racer
$ cat /Users/carolnichols/bin/script-racer
#!/bin/bash

multirust run override racer "$@"
```

And I can run that and get completions:

```
$ script-racer complete std::io::B
MATCH BufReader,47,11,/Users/carolnichols/rust/rust/src/libstd/io/buffered.rs,Struct,pub struct BufReader<R> {
MATCH BufWriter,299,11,/Users/carolnichols/rust/rust/src/libstd/io/buffered.rs,Struct,pub struct BufWriter<W: Write> {
MATCH BufRead,1252,10,/Users/carolnichols/rust/rust/src/libstd/io/mod.rs,Trait,pub trait BufRead: Read {
MATCH Broadcast,1479,11,/Users/carolnichols/rust/rust/src/libstd/io/mod.rs,Struct,pub struct Broadcast<T, U> {
MATCH Bytes,1612,11,/Users/carolnichols/rust/rust/src/libstd/io/mod.rs,Struct,pub struct Bytes<R> {
```

As soon as I move your script from script-racer to just racer, however, running `racer` from the command line hangs for me:

```
$ mv ~/bin/script-racer ~/bin/racer
$ which racer
/Users/carolnichols/bin/racer
$ racer complete std::io::B
[hangs forever]
```

Perhaps the script is calling itself?

**Comment by alexreg** [comment]
Hmm, I just tried this again and completion seems to work fine when I press `Esc`, but freezes when I press `Alt+Esc` (I then have to press `Ctrl+C` to unfreeze, i.e. cancel the shell script).

**Comment by alexreg** [comment]
Hi @carols10cents. Any update on this? Has the project been abandoned, perhaps?

**Comment by carols10cents** [comment]
Nope, it's not abandoned, I just haven't had time to spend on this lately. Feel free to fork and improve if you want though!

**Comment by alexreg** [comment]
No worries. I don't have the time either, I'm afraid. But I'd be happy to test updates. 😊
